HazelcastClientNotActiveException after job submit when using Observable

I have this job (simplified from one of our examples):

```
        Pipeline p = Pipeline.create();
        JetInstance jet = Jet.bootstrappedInstance();
        Observable<Object> observable = jet.getObservable("my-observable");
        p.readFrom(TestSources.items(1, 2, 3, 4, 5))
         .filter(number -> number % 2 == 0)
         .setName("filter out odd numbers")
         .writeTo(Sinks.observable(observable));
        UUID uuid = observable.addObserver(System.out::println);
        jet.newJob(p).join();
//        observable.removeObserver(uuid);
        Jet.shutdownAll();
```

If I submit this to the cluster I get from the jet submit

```
14:19:30.960 [ WARN] [c.h.j.i.JetClientInstanceImpl] hz.client_1 [jet] [4.2-SNAPSHOT] [4.0.1] Terminating message listener 'ce8dedf0-ef1e-4158-a35d-33923e85e2ce/__jet.observables.my-observable'. Reason: Unhandled exception, message: Client is shutting down
com.hazelcast.client.HazelcastClientNotActiveException: Client is shutting down
	at com.hazelcast.client.impl.spi.impl.ClientInvocation.notifyException(ClientInvocation.java:260)
	at com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl.shutdown(ClientInvocationServiceImpl.java:274)
	at com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl.doShutdown(HazelcastClientInstanceImpl.java:765)
	at com.hazelcast.client.impl.clientside.LifecycleServiceImpl.doShutdown(LifecycleServiceImpl.java:172)
	at com.hazelcast.client.impl.clientside.LifecycleServiceImpl.shutdown(LifecycleServiceImpl.java:150)
	at com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl.shutdown(HazelcastClientInstanceImpl.java:743)
	at com.hazelcast.jet.impl.AbstractJetInstance.shutdown(AbstractJetInstance.java:173)
	at com.hazelcast.jet.impl.JetBootstrap$InstanceProxy.shutdown(JetBootstrap.java:325)
	at org.example.JetJob.main(JetJob.java:27)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.hazelcast.jet.impl.JetBootstrap.executeJar(JetBootstrap.java:125)
	at com.hazelcast.jet.server.JetCommandLine.submit(JetCommandLine.java:209)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.hazelcast.jet.picocli.CommandLine.execute(CommandLine.java:1189)
	at com.hazelcast.jet.picocli.CommandLine.access$800(CommandLine.java:141)
	at com.hazelcast.jet.picocli.CommandLine$RunAll.handle(CommandLine.java:1416)
	at com.hazelcast.jet.picocli.CommandLine$RunAll.handle(CommandLine.java:1376)
	at com.hazelcast.jet.picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1243)
	at com.hazelcast.jet.picocli.CommandLine.parseWithHandlers(CommandLine.java:1526)
	at com.hazelcast.jet.server.JetCommandLine.runCommandLine(JetCommandLine.java:157)
	at com.hazelcast.jet.server.JetCommandLine.main(JetCommandLine.java:134)
Caused by: com.hazelcast.client.HazelcastClientNotActiveException: Client is not active.
	... 26 more
```

If I uncomment the `observable.removeObserver(uuid);` then it exits fine

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/observer/ObservableImpl.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/observer/ObservableImpl.java
@@ -16,6 +16,7 @@
 
 package com.hazelcast.jet.impl.observer;
 
+import com.hazelcast.client.HazelcastClientNotActiveException;
 import com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl;
 import com.hazelcast.config.Config;
 import com.hazelcast.config.RingbufferConfig;
@@ -266,6 +267,10 @@ public class ObservableImpl<T> implements Observable<T> {
             } else if (t instanceof HazelcastInstanceNotActiveException) {
                 if (logger.isFinestEnabled()) {
                     logger.finest("Terminating message listener '" + id + "'. Reason: HazelcastInstance is shutting down");
+                }
+            } else if (t instanceof HazelcastClientNotActiveException) {
+                if (logger.isFinestEnabled()) {
+                    logger.finest("Terminating message listener '" + id + "'. Reason: HazelcastClient is shutting down");
                 }
             } else if (t instanceof DistributedObjectDestroyedException) {
                 if (logger.isFinestEnabled()) {

